torch/jit/__init__.py

```diff
@@ -354,22 +355,29 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
-def _script_graph(fn, _frames_up=0):
+def script(fn, optimize=True, _frames_up=0):
     rcb = createResolutionCallback(_frames_up + 1)
     ast = get_jit_ast(fn)
-    return _jit_script_compile(ast, rcb)
-
-
-def script(fn, optimize=True, _frames_up=0):
-    graph = _script_graph(fn, _frames_up=_frames_up + 1)
+    schema = annotations.get_signature(fn)
+    if schema:
+        typed_def = torch._C._pack_typed_def(ast, schema[0], schema[1])
+    else:
+        typed_def = torch._C.TypedDef(ast)
+    graph = _jit_script_compile(typed_def, rcb)
     mod = ScriptModule()
     mod._create_method_from_graph('forward', graph)
+    # TODO: refactor everything so we're not 1) creating a ScriptModule
+    # 2) Throwing everything away except for the graph 3) Creating a new
+    # ScriptModule and dumping that graph in 4) Re-populating the schema
+    # because it was lost doing the previous
+    if schema:
+        mod.__getattr__('forward').set_arg_and_return_types(ast, schema[0], schema[1])
```

Comment:
Can't we pass `schema` as an argument to `_create_method_from_graph`? It seems like a good place for this info to go